src/sagemaker/jumpstart/types.py

```diff
@@ -956,6 +956,10 @@ def use_training_model_artifact(self) -> bool:
         # otherwise, return true is a training model package is not set
         return len(self.training_model_package_artifact_uris or {}) == 0
 
+    def is_gated_model(self) -> bool:
+        """Returns True if the model has a EULA key or the model bucket is gated."""
+        return self.gated_bucket or self.hosting_eula_key is not None
```

Comment:
hum, this is odd, shouldn't the pySDK only us `gated_bucket` as indicator?

Isn't it MH's unit test job to ensure that if `hosting_eula_key` is passed, then `gated_bucket` is set correctly?

Reply:
We don't have a unit test for this (tho we should). It was confusing (at least for me) to refer to gated buckets and hosting eula keys separately, so to clear the confusion i created this helper function to clarify these are the markers for gated models (if the bucket is gated or there's a eula key).